dp/generate_prsn.py

```diff
@@ -19,24 +19,36 @@
 logger = logging.getLogger(__name__)
 
 
-def dry_run(filepaths):
+def dry_run(filepaths, output_to_file=False, workdir=None):
     '''Perform metadata checks on the input files'''
-    logger.info('Dry Run')
+    if output_to_file: # Used for wps process in thunderbird
+        output_items = []
+        outputer = output_items.append
+    else:
+        outputer = logger.info
+
+    outputer('Dry Run')
     for filepath in filepaths.values():
-        logger.info('')
-        logger.info('File: {}'.format(filepath))
+        outputer('File: {}'.format(filepath))
         try:
             dataset = CFDataset(filepath)
         except Exception as e:
-            logger.exception('{}: {}'.format(e.__class__.__name__, e))
+            outputer('{}: {}'.format(e.__class__.__name__, e))
 
         for attr in 'project model institute experiment ensemble_member'.split():
             try:
-                logger.info('{}: {}'.format(attr, getattr(dataset.metadata, attr)))
+                outputer('{}: {}'.format(attr, getattr(dataset.metadata, attr)))
             except Exception as e:
-                logger.info('{}: {}: {}'.format(attr, e.__class__.__name__, e))
-        logger.info('dependent_varnames: {}'.format(dataset.dependent_varnames()))
+                outputer('{}: {}: {}'.format(attr, e.__class__.__name__, e))
+        outputer('dependent_varnames: {}'.format(dataset.dependent_varnames()))
+
+    if output_to_file:
+        filename = os.path.join(workdir, 'dry.txt')
+        with open(filename, 'w') as f:
+            for line in output_items:
+                f.write('{}\n'.format(line))
 
+        return filename
```

Comment:
I'm not sure it is a good idea for us to return something from this method in some cases but not others.  I imagine you are using the `filename` in `thunderbird` to give as a `response`.  If that is the case I think adding the path as a param will solve the issue since you will already know the output location.